### PR TITLE
Fix issue #12

### PR DESCRIPTION
**Pull request by farhadmpr**
Fix issue related to the found ticker id in the "symbols_name.json" file when name does not exist in the file, then I download ticker id from the tse site.

### CODE REVIEW

File: pytse_client/download.py [hunk 1]
```diff
@@ -23,9 +23,14 @@ def download(
 
     with ThreadPoolExecutor(max_workers=10) as executor:
         for symbol in symbols:
+            ticker_index = symbols_data.get_ticker_index(symbol)
+            if ticker_index == None:
+                print("Warning, ticker index not found, I'm trying to download it...")
```
Comment (Glyphack) [inline pytse_client/download.py:28]: I think this print is redundant because it won't affect the user.

File: pytse_client/download.py [hunk 2]
```diff
@@ -23,9 +23,14 @@ def download(
...
+                ticker_index = get_symbol_id(symbol)
+                print(ticker_index)
```
Comment (Glyphack) [inline pytse_client/download.py:30]: this is redundant too.

File: pytse_client/download.py [hunk 3]
```diff
@@ -61,6 +66,18 @@ def download_ticker_daily_record(ticker_index: str):
...
 
+def get_symbol_id(symbol_name: str):
+    url = tse_settings.TSE_SYMBOL_ID_URL.format(symbol_name + ' ')
+    response = requests_retry_session().get(url, timeout=10)
+    try:
+        response.raise_for_status()
+    except HTTPError:
+        raise Exception("Sorry, tse server did not respond")
+    
+    symbol_id = response.text.split(';')[0].split(',')[2]
```
Comment (Glyphack) [inline pytse_client/download.py:77]: There is a case here which tse might return wrong id.
for example there is no symbol named "تبر" but with this url when you pass it as symbol_name, tse will return "شبريز". so we should let user know when we cannot find id for requested symbol name.
So we should check that information tse is returning matches with our symbol_name.
Also be careful with "ک" and "ی" characters. if you pass "فملی" as symbol_name the symbol name tse returns is "فملي" so it won't match. so convert "ي" and "ك" before checking.

File: pytse_client/symbols_data.py [hunk 4]
```diff
@@ -19,8 +19,11 @@ def tickers_dict() -> Dict:
 
 
 def get_ticker_index(ticker_symbol: str):
-    return tickers_dict()[ticker_symbol]
+    tickers = tickers_dict()
+    if ticker_symbol in tickers:
+        return tickers[ticker_symbol]
+    return None
```
Comment (Glyphack) [inline pytse_client/symbols_data.py:25]:
```suggestion
    return tickers_dict().get(ticker_symbol)
```

**Comment by Glyphack** [comment]
Hi, this seems interesting. Thanks for your PR!

**Comment by farhadmpr** [comment]
Thanks for your suggestions, I solved these problems.

File: pytse_client/symbols_data.py [hunk 5]
```diff
@@ -19,8 +19,21 @@ def tickers_dict() -> Dict:
...
-    return tickers_dict()[ticker_symbol]
+    return tickers_dict().get(ticker_symbol)
 
 
 def all_symbols() -> Set:
     return set(tickers_dict().keys())
+
+
+def append_symbol_to_file(symbol_id: str, symbol_name: str):
+    new_symbol = {symbol_id.strip(): symbol_name.strip()}
```
Comment (Glyphack) [inline pytse_client/symbols_data.py:30]: Oh this should be:
```suggestion
    new_symbol = {symbol_name.strip(): symbol_id.strip()}
```

**Pull request merged**